adapter/thing/light_test.go

```diff
@@ -0,0 +1,315 @@
+						MockSetLevelSwitchBinaryState(true, errors.New("setting error"), false),
+				),
+				Nodes: []*suite.Node{
+					{
+						Name:    "set binary",
+						Command: suite.BoolMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "cmd.binary.set", "out_lvl_switch", true),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "out_lvl_switch"),
+						},
+					},
+					{
+						Name:    "wrong value type",
+						Command: suite.StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "cmd.binary.set", "out_lvl_switch", "true"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "out_lvl_switch"),
+						},
+					},
+					{
+						Name:    "wrong address",
+						Command: suite.BoolMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:3", "cmd.binary.set", "out_lvl_switch", true),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:3", "out_lvl_switch"),
+						},
+					},
+				},
+			},
+			{
+				Name: "failed set binary - binary report error",
+				Setup: routeLight(
+					mockedoutlvlswitch.NewController(t).
+						MockSetLevelSwitchBinaryState(true, nil, false).
+						MockLevelSwitchBinaryReport(true, errors.New("report error"), false),
+				),
+				Nodes: []*suite.Node{
+					{
+						Name:    "set binary",
+						Command: suite.BoolMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "cmd.binary.set", "out_lvl_switch", true),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "out_lvl_switch"),
+						},
+					},
+				},
+			},
+			{
+				Name: "failed get level - send level error",
+				Setup: routeLight(
+					mockedoutlvlswitch.NewController(t).
+						MockLevelSwitchLevelReport(99, errors.New("sending error"), false),
+				),
+				Nodes: []*suite.Node{
+					{
+						Name:    "get level",
+						Command: suite.NullMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "cmd.lvl.get_report", "out_lvl_switch"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:out_lvl_switch/ad:2", "out_lvl_switch"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	s.Run(t)
+}
+
+func TestTaskLight(t *testing.T) { //nolint:paralleltest
+	s := &suite.Suite{
+		Cases: []*suite.Case{
+			{
+				Name: "Light tasks",
+				Setup: taskLight(
+					mockedoutlvlswitch.NewController(t).
+						MockLevelSwitchLevelReport(99, nil, true).
+						MockLevelSwitchBinaryReport(false, nil, true).
+						MockLevelSwitchLevelReport(99, errors.New("task error"), false).
+						MockLevelSwitchBinaryReport(false, errors.New("task error"), false),
```

Comment:
It would be best if you mocked level report to be once 99, after that make the error report returned once as well, and then then level report being always returned at 0 (while binary respectively true and then false).

Then you should add to expectations level 0 and state false - and by running the test you will be sure that everything happened including the errors.